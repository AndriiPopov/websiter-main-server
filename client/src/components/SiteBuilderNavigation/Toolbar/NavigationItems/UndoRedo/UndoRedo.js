import React from 'react'
import { ActionCreators as UndoActionCreators } from 'redux-undo'
import { connect } from 'react-redux'

import AUX from '../../../../../hoc/Aux';
import NavigationItemButton from '../NavigationItemButton/NavigationItemButton';

export const UndoRedo = ({ canUndo, canRedo, onUndo, onRedo }) => (
  <AUX>
        <NavigationItemButton 
            clicked={onUndo} 
            disabled={!canUndo}
            icon='<svg height="24" viewBox="0 0 24 24" width="24"><path d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"></path></svg>' />
        <NavigationItemButton 
            clicked={onRedo} 
            disabled={!canRedo}
            icon='<svg width="24" height="24" viewBox="0 0 24 24"><path d="M18.4 10.6C16.55 8.99 14.15 8 11.5 8c-4.65 0-8.58 3.03-9.96 7.22L3.9 16c1.05-3.19 4.05-5.5 7.6-5.5 1.95 0 3.73.72 5.12 1.88L13 16h9V7l-3.6 3.6z"></path></svg>' />
    </AUX>
);

const mapStateToProps = state => ({
  canUndo: state.builder.past.length > 0,
  canRedo: state.builder.future.length > 0
})

const mapDispatchToProps = ({
  onUndo: UndoActionCreators.undo,
  onRedo: UndoActionCreators.redo
})
export default connect(mapStateToProps, mapDispatchToProps)(UndoRedo);