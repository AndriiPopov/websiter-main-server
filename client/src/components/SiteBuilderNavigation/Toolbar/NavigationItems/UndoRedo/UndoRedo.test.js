import React from 'react';
import {mount, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { UndoRedo } from './UndoRedo';

configure({adapter: new Adapter()});

describe('UndoRedo', () => {
    it('should be defined', () => {
        expect(UndoRedo).toBeDefined();
    });

    it('it should render button with onclick null when disabled', () => {
        const tree = mount(
            <UndoRedo
                disabled
                clicked="clicked"
                icon="<svg></svg>" />
        );
        expect(toJson(tree)).toMatchSnapshot();
    });
});