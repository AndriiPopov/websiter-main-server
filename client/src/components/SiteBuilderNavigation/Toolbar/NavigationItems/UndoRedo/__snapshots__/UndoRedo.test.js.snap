// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UndoRedo it should render button with onclick null when disabled 1`] = `
<UndoRedo
  clicked="clicked"
  disabled={true}
  icon="<svg></svg>"
>
  <aux>
    <NavigationItemButton
      disabled={true}
      icon="<svg height=\\"20\\" viewBox=\\"0 0 24 24\\" width=\\"20\\"><path d=\\"M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z\\"></path></svg>"
    >
      <li
        className="NavigationItemButton disabled"
        onClick={null}
      >
        <Svg
          icon="<svg height=\\"20\\" viewBox=\\"0 0 24 24\\" width=\\"20\\"><path d=\\"M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z\\"></path></svg>"
        >
          <svg
            height="20"
            viewBox="0 0 24 24"
            width="20"
          >
            <path
              d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"
              key="0"
            />
          </svg>
        </Svg>
      </li>
    </NavigationItemButton>
    <NavigationItemButton
      disabled={true}
      icon="<svg height=\\"20\\" viewBox=\\"0 0 24 24\\" width=\\"20\\"><path d=\\"M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z\\"></path></svg>"
    >
      <li
        className="NavigationItemButton disabled"
        onClick={null}
      >
        <Svg
          icon="<svg height=\\"20\\" viewBox=\\"0 0 24 24\\" width=\\"20\\"><path d=\\"M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z\\"></path></svg>"
        >
          <svg
            height="20"
            viewBox="0 0 24 24"
            width="20"
          >
            <path
              d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"
              key="0"
            />
          </svg>
        </Svg>
      </li>
    </NavigationItemButton>
  </aux>
</UndoRedo>
`;
